connectgateway/get_namespace.py

```diff
@@ -0,0 +1,106 @@
+    try:
+        client_options = {}
+        if location != "global":
+            # If the location is not global, the endpoint needs to be set to the regional endpoint.
+            regional_endpoint = f"{location}-connectgateway.googleapis.com"
+            client_options = {"api_endpoint": regional_endpoint}
+        gateway_client = gateway_v1.GatewayControlClient(client_options=client_options)
+        request = gateway_v1.GenerateCredentialsRequest()
+        request.name = membership_name
+        response = gateway_client.generate_credentials(request=request)
+        print(f'GKE Connect Gateway Endpoint: {response.endpoint}')
+        if not response.endpoint:
+            print("Error: GKE Connect Gateway Endpoint is empty.")
+            sys.exit(1)
+        return response.endpoint
+    except exceptions.NotFound as e:
+        print(f'Membership not found: {e}')
+        sys.exit(1)
+    except Exception as e:
+        print(f'Error fetching GKE Connect Gateway URL: {e}')
+        sys.exit(1)
+
+
+def configure_kubernetes_client(gateway_url: str, service_account_key_path: str) -> client.CoreV1Api:
+    """Configures the Kubernetes client with the GKE Connect Gateway URL and credentials."""
+
+    configuration = client.Configuration()
+
+    # Configure the API client with the custom host.
+    configuration.host = gateway_url
+
+    # Set the authentication header with the GCP OAuth token
+    if not service_account_key_path:
+        print("Error: SERVICE_ACCOUNT_KEY_PATH environment variable not set.")
+        sys.exit(1)
+    try:
+        credentials = service_account.Credentials.from_service_account_file(
+            service_account_key_path, scopes=SCOPES
+        )
+    except FileNotFoundError:
+        print(f"Error: Service account key file not found at {service_account_key_path}")
+        sys.exit(1)
+    except Exception as e:
+        print(f"Error loading service account credentials: {e}")
+        sys.exit(1)
+    auth_req = requests.Request()
+    credentials.refresh(auth_req)
+    configuration.api_key = {'authorization': f'Bearer {credentials.token}'}
+
+    api_client = client.ApiClient(configuration=configuration)
+    return client.CoreV1Api(api_client)
+
+
+def get_default_namespace(api_client: client.CoreV1Api):
+    """Get default namespace in the Kubernetes cluster."""
+    try:
+        namespace = api_client.read_namespace(name="default")
+        return namespace
+    except client.ApiException as e:
+        print(f"Error getting default namespace: {e}\nStatus: {e.status}\nReason: {e.reason}")
+        sys.exit(1)
+
+
+def get_namespace(membership_name: str, location: str, service_account_key_path: str):
+    """Main function to connect to the cluster and get the default namespace."""
+    gateway_url = get_gateway_url(membership_name, location)
+    core_v1_api = configure_kubernetes_client(gateway_url, service_account_key_path)
+    return get_default_namespace(core_v1_api)
```

Comment:
lint: samples should process the result, e.g. printing to stdout. https://googlecloudplatform.github.io/samples-style-guide/#result

The return statement can be placed after the "END" region tag to ensure the object is returned to help with testing.